realsense2_camera/src/ros_sensor.cpp

```diff
@@ -136,8 +136,8 @@ void RosSensor::UpdateSequenceIdCallback()
             {
                 set_option(RS2_OPTION_SEQUENCE_ID, parameter.get_value<int>());
                 std::vector<std::function<void()> > funcs;
-                funcs.push_back([this](){set_sensor_parameter_to_ros(RS2_OPTION_GAIN);});
-                funcs.push_back([this](){set_sensor_parameter_to_ros(RS2_OPTION_EXPOSURE);});
+                funcs.push_back([this](){set_sensor_parameter_to_ros<int>(RS2_OPTION_GAIN);});
+                funcs.push_back([this](){set_sensor_parameter_to_ros<int>(RS2_OPTION_EXPOSURE);});
```

Comment:
Unfortunately this cannot be tested with rosbag testcase, since the depth controls (such as exposure, etc.,)  are not exposed for rosbag scenario. 
So, I have added a live camera test and verified locally with D455 that it is failing in 'ros2-development' branch and passing with this PR.